docs/getting_started/user-guide.rst

```diff
@@ -1228,6 +1260,63 @@ it could be that a different method of choosing or sorting tags (beyond the defa
 and filter) is needed.
 
 
+Sync Registry
+-------------
+
+A sync is when we take your local filesystem registry, and retrieve updates from the remote defined at 
+``sync_registry`` in your settings.yaml. Since sync will be writing recipes to the filesystem
+it only works if you target a filesystem registry (meaning that the default registry
+as a remote will not work). 
+
+.. note::
+
+   By default, the first filesystem registry found in your settings under the registry list will
+   be used. To provide a one off registry folder (that should exist but does not need to be in your
+   defined list) you can use ``--registry``.
+
+As an example, if we do this without changing the defaults:
+
+.. code-block:: console
+
+    $ shpc sync-registry
+    This command is only supported for a filesystem registry! Add one or use --registry.    
+
+We can then make a dummy directory to support sync. You could also make this directory and add to your settings proper under ``registry``.
+
+.. code-block:: console
+
+    $ mkdir -p ./registry
+    $ shpc sync-registry --registry ./registry
+
+Will compare your main registry folder against the main branch and only add new recipes 
+that you do not have. To ask to update from a specific reference (tag or branch):
+
+.. code-block:: console
+
+    $ shpc sync-registry --registry ./registry --tag 0.0.58
+    
+You can also ask to add just a specific container:
+
+.. code-block:: console
+
+    $ shpc sync-registry --registry ./registry quay.io/not-local/container
+        
+And finally, you can ask to add new containers and completely update container.yaml files. 
+
+.. code-block:: console
+
+    $ shpc sync-registry --registry ./registry --all
+    
+This means we do a side by side comparison of your filesystem registry and the upstream, and we add new
+recipes folders that you don't have, and we replace any upstream files into recipes that you do have.
+Be careful with this option, as if you've made changes to a container.yaml or associated
+file in the upstream they will be lost. For this reason, we always recommend that you do a dry run first:
```

Comment:
In a different test, it told me it would add some git object files:
```
Cloning into '/tmp/shpc-tmp.e1ue2gm4'...
remote: Enumerating objects: 68, done.
remote: Counting objects: 100% (68/68), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 68 (delta 1), reused 52 (delta 1), pack-reused 0
Unpacking objects: 100% (68/68), done.
.git/objects/46 will be added newly.
.git/objects/b8 will be added newly.
.git/objects/6d will be added newly.
.git/objects/5b will be added newly.
.git/objects/40 will be added newly.
.git/objects/19 will be added newly.
.git/objects/28 will be added newly.
.git/objects/44 will be added newly.
.git/objects/83 will be added newly.
.git/objects/30 will be added newly.
.git/objects/38 will be added newly.
.git/objects/98 will be added newly.
```
I'd expect shpc to sync everything but `.git`. If people want the local registry to be a git checkout, _they_ have to deal with it (and do the fetch, commit, etc, themselves)

Reply:
Oh that seems like a bug, because it should not be adding anything aside from the container recipes! I'll put in a fix for this tonight after work.

Reply:
Can you give me the command to reproduce this? I want to see if there is a way to only include paths with container.yaml.

Reply:
I can't reproduce this one precisely. But I've got something similar. On a clean checkout with no local modifications
```
$ rm -rf test_sync/
$ git clone git@github.com:singularityhub/singularity-hpc.git test_sync
$ cd test_sync/
$ git remote remove origin
$ cd ..
$ shpc config add registry $PWD/test_sync
$ shpc sync-registry
(...)
adminer will be added newly.
.git/logs/refs/remotes/origin will be added newly.
.git/refs/remotes/origin will be added newly.
```